src/tdx.rs

```diff
@@ -0,0 +1,143 @@
+use crate::types::enclave_identity::QeTcbStatus;
+use crate::types::report::{Td10ReportBody, Td15ReportBody};
+use crate::types::tcb_info::{TcbInfo, TcbStatus, TdxModuleIdentity};
+
+pub fn verify_tdx_module(
```

Comment:
Any reason of doing this in functional manner ?

Reply:
I don't understand the question. But I thought moving the implementation here is better in terms of code organization, as opposed to making it a `TcbInfo` impl method as done previously. 

Also, I have separated out a `find_tdx_module_identity` internal function to be re-used by the `check_for_relaunch` function.